### DISCUSSION TITLE
Multi Tenancy for Development Teams

### DISCUSSION THREAD

**Discussion by disposab1e** [argoproj/argo-workflows#6337] [Q&A]
What is the recommended approach for multi tenancy? Is it possible or recommended to install dedicated namespaced Argo Workflow controllers for different development teams?

**Comment by disposab1e** [comment]
Actually with have dedicated ArgoCD installations for different Development Teams in the same cluster, so they can install, manage and use ArgoCD in a "team scope". Now we're thinking about adding the possibility to do the same with Argo Events, Workflows and Rollouts. Just searching for the best way to do it....

**Reply by tooptoop4** [reply]
@disposab1e what did u end up doing? my current thinking is cluster admin defines team1 with namespace1/serviceaccount1/rolebinding1/gitrepo1, and defines team2 with namespace2/serviceaccount2/rolebinding2/gitrepo2. But what prevents team2 from creating a workflow yaml that references namespace1/serviceaccount1 ?
